src/main/scala/diplomacy/AddressRange.scala

```diff
@@ -53,6 +53,7 @@ object AddressRange
         case Some(z) => z :: tail
         case None => x :: head :: tail
       }
+      case (head, tail) => head ++ Seq(tail)
```

Comment:
Pretty sure this should just be: `case (Nil, _) => require(false)`

Reply:
What I don't like is your PR seems to suggest this case is possible and that there is something to be done.

Reply:
> Pretty sure this should just be: `case (Nil, _) => require(false)`

How about this, to get rid of the warning?
```
case (_, _) => require(false); Nil
```

Reply:
So I just spent some time googling for this and playing around with scala, because I have often wondered how best to mark something unreachable. The problem with 'require(false)' is that it does not have polymorphic return type. That's because, usually, a require can succeed. This is why you added '; Nil'.

There are some ways to get a polymorphic return type in scala that I found:
- you can `throw` something, perhaps `AssertionError("unreachable")`
- you can put `???` which results in [error] Caused by: scala.NotImplementedError: an implementation is missing

Having to write `throw AssertionError("unreachable") ` is more typing than `??? // unreachable`, but it is more descriptive.

Frankly, I would like to `def unreachable: Empty = throw AssertionError("unreachable")` somewhere in rocket-chip and then we can just put `unreachable` in these places.